azure-pipelines/release.yml

```diff
@@ -1,131 +1,272 @@
-              }
-            displayName: 🚀 Publish to Marketplace
-            workingDirectory: $(Pipeline.Workspace)
-            env:
-              VSCE_PAT: $(VSCodeMarketplacePAT)
-- stage: 'TagRelease'
-  displayName: 'Tag release of vscode-csharp'
-  dependsOn: 'PublishStage'
-  condition: succeeded('PublishStage')
-  jobs:
-  - job: 'Tag'
+resources:
+  repositories:
+  - repository: 1ESPipelineTemplates
+    type: git
+    name: 1ESPipelineTemplates/1ESPipelineTemplates
+    ref: refs/tags/release
+  pipelines:
+  - pipeline: officialBuildCI
+    source: dotnet-vscode-csharp
+    branch: main
+extends:
+  template: v1/1ES.Official.PipelineTemplate.yml@1ESPipelineTemplates
+  parameters:
     pool:
-      vmImage: ubuntu-latest
-    steps:
-    - task: NodeTool@0
-      displayName: 'Install Node.js 18.x'
-      inputs:
-        versionSpec: '18.x'
-    - checkout: self
-      clean: true
-      submodules: true
-      fetchTags: false
-      fetchDepth: 0
-    - pwsh: |
-        git checkout $(resources.pipeline.officialBuildCI.sourceCommit)
-      displayName: 'Checkout build source branch'
-    - pwsh: |
-        npm ci
-        npm install
-        npm install -g gulp
-        gulp installDependencies
-      displayName: 'Install dependencies.'
-    - pwsh: |
-        # build number is generated by nerdbank git versioning and set to AzureDev
-        gulp createTags --releaseVersion $(resources.pipeline.officialBuildCI.runName) --releaseCommit $(resources.pipeline.officialBuildCI.sourceCommit) --dryRun ${{ parameters.test }}
-      env:
-        GitHubPAT: $(BotAccount-dotnet-bot-content-rw-grained-pat)
-      displayName: 'Create release tags'
+      name: netcore1espool-internal
+      image: 1es-windows-2022-pt
+      os: windows
+    customBuildTags:
+    - ES365AIMigrationTooling
+    stages:
+    - stage: PublishStage
+      jobs:
+      - deployment: PublishToMarketplace
+        displayName: PublishToMarketplace
+        environment: vscode-csharp-release-approvals
+        pool:
+          name: netcore1espool-internal
+          image: 1es-ubuntu-2204-pt
+          os: linux
+        strategy:
+          runOnce:
+            deploy:
+              steps:
+              - download: 'none'
+              - task: DownloadPipelineArtifact@2
+                displayName: '📦 Download artifacts from build pipeline.'
+                inputs:
+                  buildType: 'specific'
+                  project: 'internal'
+                  definition: 1264
+                  buildVersionToDownload: 'specific'
+                  buildId: '$(resources.pipeline.officialBuildCI.runID)'
+                  branchName: '$(resources.pipeline.officialBuildCI.sourceBranch)'
+              - pwsh: |
+                  npm install --global vsce
+                displayName: 'Install vsce'
+              - pwsh: |
+                  # Choose whether to upload to prerelease or release based on the artifacts generated by the build pipeline.
```

Comment:
Release pipeline now automatically detects the correct channel based on the build artifact name